archilogic-com/3dio-js
# Data3d cache

Scope & Features:
- adds promise based cache to data3d files:
- improves performance by reusing request promises to same URL

Design:
- uses `promise-cache.js` module
- a-frame furniture component had to be updated so that in case of changeable materials a clone of the data3d object is being used. otherwise all the furniture with the same data3d URL would change as they are sharing data3d objects for reasons of memory optimization.

How to use:
- no special user interaction needed.
- loading furniture should behave the same way as without cache - just faster ;)

How to test:
1. start local dev server `npm run start`
2. open a-frame test case `localhost:8080/examples-browser/a-frame/`
3. reload a few times
4. colored chairs should always have the same color after reload
